javascript/modules/client/src/internal/interfaces/plugins.ts

```diff
@@ -152,33 +122,171 @@ export type ProposalCreationStepValue =
   | { key: ProposalCreationSteps.CREATING; txHash: string }
   | { key: ProposalCreationSteps.DONE; proposalId: string };
 
+// PROPOOSAL VOTING
+export enum VoteProposalStep {
+  VOTING = "voting",
+  DONE = "done",
+}
+
+export type VoteProposalStepValue =
+  | { key: VoteProposalStep.VOTING; txHash: string }
+  | { key: VoteProposalStep.DONE; voteId: string };
+
+// PROPOOSAL EXECUTION
+export enum ExecuteProposalStep {
+  EXECUTING = "executing",
+  DONE = "done",
+}
+
+export type ExecuteProposalStepValue =
+  | { key: ExecuteProposalStep.EXECUTING; txHash: string }
+  | { key: ExecuteProposalStep.DONE };
+
+// VOTING CONFIGURATION
+export enum SetPluginConfigStep {
+  CREATING_PROPOSAL = "creating_proposal",
+  DONE = "done",
+}
+
+export type SetPluginConfigStepValue =
+  | { key: SetPluginConfigStep.CREATING_PROPOSAL; txHash: string }
+  | { key: SetPluginConfigStep.DONE; };
+
 // PROPOSAL RETRIEVAL
 
-export type Erc20Proposal = Proposal & {
-  voteId: string;
-  token: Erc20Token;
+// Long version
+export type ProposalBase = {
+  id: string
+  dao: {
+    address: string,
+    name: string,
+  },
+  creatorAddress: string;
+  metadata: ProposalMetadata;
+  startDate: Date;
+  endDate: Date;
+  creationDate: Date;
+  actions: Array<ProposalAction>;
+  status: ProposalStatus;
+}
 
-  result: {
-    yea?: number;
-    nay?: number;
-    abstain?: number;
-  };
+export type Erc20Proposal = ProposalBase & {
+  result: Erc20ProposalResult;
+  settings: ProposalVotingSettings;
+  token: Erc20TokenDetails;
+  usedVotingWeight: bigint;
+  votes: Array<{ address: string; vote: VoteValues; weight: bigint }>;
+}
 
-  config: {
-    participationRequiredPct: number;
-    supportRequiredPct: number;
+export type AddressListProposal = ProposalBase & {
+  result: AddressListProposalResult;
+  settings: ProposalVotingSettings;
+  votes: Array<{ address: string; vote: VoteValues; }>;
+};
+
+/**
+ * Contains the human-readable information about a proposal
+ */
+ export type ProposalMetadata = {
+  title: string;
+  summary: string;
+  description: string;
+  resources: Array<{ url: string; name: string }>;
+  media: {
+    header?: string;
+    logo?: string;
   };
+};
+
+/**
+ * Contains the human-readable information about a proposal
+ */
+export type ProposalAction = {
+  to: string;
+  value: bigint;
+  data: Uint8Array;
+};
```

Comment:
There is also the `DaoAction`. Please, let's only use one of them and remove the duplicate.